test/unit/query-interface/upsert.test.ts

```diff
@@ -36,8 +36,8 @@ describe('QueryInterface#upsert', () => {
     const firstCall = stub.getCall(0);
     expectsql(firstCall.args[0] as string, {
       default: 'INSERT INTO [Users] ([firstName]) VALUES ($sequelize_1) ON CONFLICT ([id]) DO UPDATE SET [firstName]=EXCLUDED.[firstName];',
-      mariadb: 'INSERT INTO `Users` (`firstName`) VALUES ($sequelize_1) ON DUPLICATE KEY UPDATE `firstName`=VALUES(`firstName`);',
-      mysql: 'INSERT INTO `Users` (`firstName`) VALUES ($sequelize_1) ON DUPLICATE KEY UPDATE `firstName`=VALUES(`firstName`);',
+      mariadb: 'INSERT INTO `Users` (`firstName`) VALUES ($sequelize_1) ON DUPLICATE KEY UPDATE `firstName`=$sequelize_1;',
+      mysql: 'INSERT INTO `Users` (`firstName`) VALUES ($sequelize_1) ON DUPLICATE KEY UPDATE `firstName`=$sequelize_1;',
```

Comment:
It's likely that we'll eventually add a `Model.bulkUpsert` method, which would generate something like this:

```sql
INSERT INTO `Users` (`firstName`) VALUES ($sequelize_1),($sequelize_2) ON DUPLICATE KEY UPDATE `firstName`=VALUES(`firstName`);
```

In that scenario, using `VALUES(firstName)` is important as it will return a different value for each row.

I think a way to reconcile both would be to have the upsert query be generated using `VALUES(...)` by default, but use the value itself if the value if an instanceof `SequelizeMethod` (for literals).

Basically,

```js
const [insertUser] = await User.upsert({
  counter: 10
});
```

would use `ON DUPLICATE KEY UPDATE counter = VALUES(counter)`,

but 

```js
const [insertUser] = await User.upsert({
  counter: literal('counter + 1')
});
```

would use `ON DUPLICATE KEY UPDATE counter = counter + 1`,